lib/pbench/server/api/resources/query_apis/month_indices.py

```diff
@@ -1,92 +1,43 @@
         Report the month suffixes for run data indices.
 
-        NOTE: No authorization or input parameters are required for this API.
-
-        Returns a list of "YYYY-mm" date strings corresponding to the months in
-        which tarballs were indexed into the appropriate run-data index. (E.g.,
-        drb.v6.run-data.2020-11). This list is in DESCENDING order:
-
-        [
-            "2020-12",
-            "2020-11",
-            "2020-04"
-        ]
+        NOTE: No authorization or input parameters are required for this API,
+        and there is no JSON query to form.
         """
-        self.logger.info(
-            "Discover months for run-data index prefix {}", self.prefix,
-        )
-
-        uri = f"{self.es_url}/_aliases"
-        try:
-            # query Elasticsearch
-            es_response = requests.get(uri, headers={"Accept": "application/json"})
-            es_response.raise_for_status()
-        except requests.exceptions.HTTPError as e:
-            self.logger.exception("HTTP error {} from Elasticsearch post request", e)
-            abort(502, message="INTERNAL ERROR")
-        except requests.exceptions.ConnectionError:
-            self.logger.exception(
-                "Connection refused during the Elasticsearch post request"
-            )
-            abort(502, message="Network problem, could not post to Elasticsearch")
-        except requests.exceptions.Timeout:
-            self.logger.exception(
-                "Connection timed out during the Elasticsearch post request"
-            )
-            abort(504, message="Connection timed out, could not post to Elasticsearch")
-        except requests.exceptions.InvalidURL:
-            self.logger.exception(
-                "Invalid url {} during the Elasticsearch post request", uri
-            )
-            abort(500, message="INTERNAL ERROR")
-        except Exception:
-            self.logger.exception(
-                "Exception occurred during the Elasticsearch post request"
-            )
-            abort(500, message="INTERNAL ERROR")
-        else:
-            months = []
-            target = f"{self.prefix}.v6.run-data."
-            try:
-                es_json = es_response.json()
-                self.logger.info("looking for {} in {}", target, es_json)
-                for index in es_json.keys():
-                    if target in index:
-                        months.append(index.split(".")[-1])
-                months.sort(reverse=True)
-                self.logger.info("found months {!r}", months)
-            except KeyError:
-                self.logger.exception("ES response not formatted as expected")
-                abort(500, message="INTERNAL ERROR")
-            else:
-                # construct response object
-                return jsonify(months)
+        return {"path": "/_aliases"}
+
+    def postprocess(self, es_response: Response):
+        months = []
+        target = f"{self.prefix}.v6.run-data."
```

Comment:
I think you can define a class attribute on `ElasticBase` to hold the `v6.run-data` string.  It would be accessible for use here, and `_gen_month_range()` could access it directly instead of receiving it as a parameter.  

Then it would be defined only once, instead of replicated all over the place, and it would be much easier to deal with when it's time to make it shared/dynamic.

Reply:
All of the index name/template handling will be changed by my next PR, and I'm not going to worry about over-optimizing here.